lib/pbench/server/api/resources/query_apis/query_dataset_list.py

```diff
@@ -0,0 +1,290 @@
+            }
+        ]
+        """
+        json_data = request.get_json(silent=True)
+        if not json_data:
+            self.logger.info("Invalid JSON object. Query: {}", request.url)
+            abort(400, message="Missing request payload")
+
+        try:
+            user = json_data["user"]
+            controller = json_data["controller"]
+            start_arg = json_data["start"]
+            end_arg = json_data["end"]
+        except KeyError:
+            keys = [
+                k for k in ("user", "controller", "start", "end") if k not in json_data
+            ]
+            self.logger.info("Missing required JSON keys {}", ",".join(keys))
+            abort(400, message=f"Missing request data: {','.join(keys)}")
+
+        # We need to support a query for public data without requiring
+        # authorization; however if an Authorization header is given,
+        # it must be of recommended JWT "Bearer" schema, and have a
+        # non-empty token.
+        #
+        # TODO: validate the token with the user management object.
+        authorization = request.headers.get("Authorization")
+        session_token = None
+        if authorization:
+            type, token = authorization.split()
+            if type.lower() == "bearer":
+                session_token = token
+
+            if not session_token:
+                self.logger.warn(
+                    '"Authorization" header specifies unsupported or missing '
+                    ' schema; use "Authorization: Bearer <JWT-token>"'
+                )
+                abort(401, message="invalid user authorization")
+
+        try:
+            start = parser.parse(start_arg).replace(day=1)
+            end = parser.parse(end_arg).replace(day=1)
+        except Exception as e:
+            self.logger.info(
+                "Invalid start or end time string: {}, {}: {}", start_arg, end_arg, e
+            )
+            abort(400, message="Invalid start or end time string")
+
+        # We have nothing to authorize against yet, but print the specified
+        # user and session token to prove we got them (and so the variables
+        # aren't diagnosed as "unused")
+        self.logger.info(
+            "QueryControllers POST for user {}, prefix {}, session {}: ({}: {} - {})",
+            user,
+            self.prefix,
+            session_token,
+            controller,
+            start,
+            end,
+        )
+
+        payload = {
+            "_source": {
+                "includes": [
+                    "@metadata.controller_dir",
+                    "@metadata.satellite",
+                    "run.controller",
+                    "run.start",
+                    "run.end",
+                    "run.name",
+                    "run.config",
+                    "run.prefix",
+                    "run.id",
+                ]
+            },
+            "sort": {"run.end": {"order": "desc"}},
```

Comment:
Why do we sort by ending timestamp when we add a field to the results (line 273) with the _beginning_ timestamp?